core/vm/contracts_test.go

```diff
@@ -142,21 +143,29 @@ func benchmarkPrecompiled(addr string, test precompiledTest, bench *testing.B) {
 
 	bench.Run(fmt.Sprintf("%s-Gas=%d", test.Name, contract.Gas), func(bench *testing.B) {
 		bench.ReportAllocs()
+		start := time.Now().Nanosecond()
 		bench.ResetTimer()
 		for i := 0; i < bench.N; i++ {
 			contract.Gas = reqGas
 			copy(data, in)
 			res, err = RunPrecompiledContract(p, data, contract)
 		}
 		bench.StopTimer()
+		elapsed := float64(time.Now().Nanosecond() - start)
+		if elapsed < 1 {
+			elapsed = 1
+		}
+		elapsed = elapsed / 100000000
+		gasUsed := reqGas * uint64(bench.N) / 1000 / 1000
```

Comment:
You can let some of those zeroes cancel eachother, and later on just multiply by 1000 to get mgas/s. Something like this: 
`gasUsed := reqGas * uint64(bench.N)`
`bench.ReportMetric(float64(1000 * gasUsed)/elapsed, "mgas/s")`